config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +110,146 @@ const (
+	// When omitted, this means the user has no opinion and the platform is left
+	// to choose reasonable defaults. These defaults are subject to change over time.
+	// The current default is `kubernetes.io/os: linux` so that Pods can be scheduled onto any available node.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// tolerations is a list of tolerations applied to metrics server components
+	// tolerations is optional.
+	//
+	// When omitted, this means the user has no opinion and the platform is left
+	// to choose reasonable defaults. These defaults are subject to change over time.
+	// Maximum length for this list is 10
+	// +kubebuilder:validation:MaxItems=10
+	// +optional
+	Tolerations []v1.Toleration `json:"tolerations,omitempty"`
+
+	// resources defines the compute resource requests and limits for the Metrics Server container.
+	// This includes CPU, memory and HugePages constraints to help control scheduling and resource usage.
+	// When not specified, defaults are used by the platform. Requests cannot exceed limits.
+	// Resources is optional.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// This is a simplified API that maps to Kubernetes ResourceRequirements.
+	// +optional
+	Resources *MetricServerConfigContainerResources `json:"resources,omitempty"`
+
+	// topologySpreadConstraints defines rules for how Metrics Server Pods should be distributed
+	// across topology domains such as zones, nodes, or other user-defined labels.
+	// topologySpreadConstraints is optional.
+	// This helps improve high availability and resource efficiency by avoiding placing
+	// too many replicas in the same failure domain.
+	//
+	// When omitted, this means no opinion and the platform is left to choose a default, which is subject to change over time.
+	// This field maps directly to the `topologySpreadConstraints` field in the Pod spec.
+	// Maximum length for this list is 10
+	// +kubebuilder:validation:MaxItems=10
+	// +optional
+	TopologySpreadConstraints []v1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
+}
+
+// Audit defines the configuration for Metrics Server audit logging.
+type Audit struct {
+	// profile specifies the audit log level to use.
+	// Valid values are:
+	// - "metadata" - log metadata about requests (default)
+	// - "request" - log metadata and request payloads
+	// - "requestresponse" - log metadata, requests, and responses
+	// - "none" - don't log requests
+	// The default audit log level is "metadata"
+	//
+	// See: https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#audit-policy
+	// for more details about audit logging.
+	//
+	// +kubebuilder:validation:Enum=metadata;request;requestresponse;none
+	// +optional
+	Profile AuditProfileType `json:"profile,omitempty"`
+}
+
+// AuditProfileType defines the audit policy profile type.
+// +kubebuilder:validation:Enum=none;metadata;request;requestresponse
```

Comment:
Don't put the enum marker on both the type and the field - pick one or the other. Using both causes an `allOf` in the OpenAPI schema for the field (see https://github.com/openshift/api/pull/2322/files#diff-2a283f1c7a9ed75159eeb51f06ea88be271d34a335475d4155ea8d8ee364fcb5R66-R76 ) which can end up causing weird behaviors if any drift happens.